components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+                    register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_INBOUND_ENDPOINT);
+
+        }
+    }
+
+    @Override
+    public void incrementCount(String metricName, Map<String, String[]> properties) {
+        Counter counter = (Counter) metricMap.get(metricName);
+        String[] value = properties.get(metricName);
+
+        counter.labels(value).inc();
+    }
+
+    @Override
+    public void decrementCount(String metricName, Map<String, String> properties) {
+
+    }
+
+    @Override
+    public Object getTimer(String metricName, Map<String, String[]> properties) {
+        String[] value = properties.get(metricName);
+        Histogram timer = (Histogram) metricMap.get(metricName);
+
+        return timer.labels(value).startTimer();
+    }
+
+    @Override
+    public void observeTime(Object timer) {
+        Histogram.Timer timer2 = (Histogram.Timer) timer;
+        timer2.observeDuration();
+    }
+
+    @Override
+    public void serverUp(String host, String port, String javaVersion, String javaHome) {
+        Gauge gauge = (Gauge) metricMap.get("wso2_integration_server_up");
+        gauge.labels(host, port, javaHome, javaVersion).set(System.currentTimeMillis());
+    }
+
+    @Override
+    public void serverDown(String host, String port, String javaVersion, String javaHome) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVER_UP);
+        gauge.labels(host, port, javaHome, javaVersion).set(0);
+    }
+
+    @Override
+    public void serviceUp(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(System.currentTimeMillis());
+        System.out.println(gauge.labels(serviceName, serviceType).get());
+        System.out.println(System.currentTimeMillis());
+    }
+
+    @Override
+    public void serviceDown(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(0);
+    }
+
+    @Override
+    public void initMetrics() {
+        PrometheusMetricCreator prometheusMetricCreator = new PrometheusMetricCreator();
+        prometheusMetricCreator.createProxyServiceMetric();
+        prometheusMetricCreator.createAPIServiceMetric();
+        prometheusMetricCreator.createInboundEndpointMetric();
+        prometheusMetricCreator.createProxyServiceErrorMetric();
+        prometheusMetricCreator.createApiErrorMetric();
+        prometheusMetricCreator.createInboundEndpointErrorMetric();
+
+        prometheusMetricCreator.createServerUpMetrics();
+        prometheusMetricCreator.createServiceUpMetrics();
+    }
+
+    enum SERVICE {
+        PROXY,
+        API,
+        INBOUND_ENDPOINT
+    }
+
+    /**
+     * Instrument metrics on server deployment.
+     *
+     * @param configs The map of configs defined in the deployment.toml file
```

Comment:
Shall we add the missing javadoc comment here

Reply:
Fixed